src/lib/Liblog/pbs_log.c

```diff
@@ -724,15 +725,15 @@ log_err(int errnum, const char *routine, const char *text)
 		LPVOID	lpMsgBuf;
 		int	err = GetLastError();
 		int		len;
-
+		snprintf(buf, LOG_BUF_SIZE, "GetLastError() returned: %d. ", err);
 		FormatMessage(
 			FORMAT_MESSAGE_ALLOCATE_BUFFER |
 			FORMAT_MESSAGE_FROM_SYSTEM |
 			FORMAT_MESSAGE_IGNORE_INSERTS,
 			NULL, err,
 			MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
 			(LPTSTR)&lpMsgBuf, 0, NULL);
-		strncpy(buf, lpMsgBuf, sizeof(buf));
+		strncat(buf, lpMsgBuf, LOG_BUF_SIZE - (int)strlen(buf));
```

Comment:
Should this line be strncat(buf, lpMsgBuf, LOG_BUF_SIZE - strlen(log_buffer) - 1)) ?